### PR TITLE
Use `ParamVector` everywhere

### PR DESCRIPTION
**Pull request by JCGoran**
Also refactored some functions so they don't use `char*` but a `std::string` instead.

### CODE REVIEW

File: src/codegen/codegen_cpp_visitor.cpp [hunk 1]
```diff
@@ -32,6 +32,20 @@ bool CodegenCppVisitor::ion_variable_struct_required() const {
...
+    std::string str;
+    bool is_first = true;
+    for (const auto& param: params) {
+        if (is_first) {
+            is_first = false;
+        } else {
+            str += ", ";
+        }
+        str += std::get<3>(param);
+    }
```
Comment (1uc) [inline src/codegen/codegen_cpp_visitor.cpp:45]: You can use `fmt::join`.

File: src/codegen/codegen_neuron_cpp_visitor.cpp [hunk 2]
```diff
@@ -31,6 +31,7 @@ using visitor::RenameVisitor;
 using visitor::VarUsageVisitor;
 
 using symtab::syminfo::NmodlType;
+using ParamVector = std::vector<std::tuple<std::string, std::string, std::string, std::string>>;
```
Comment (1uc) [inline src/codegen/codegen_neuron_cpp_visitor.cpp:34]: Please remove this. `ParamVector` should be correctly defined in a header.

File: src/codegen/codegen_coreneuron_cpp_visitor.cpp [hunk 3]
```diff
@@ -39,6 +39,8 @@ using visitor::RenameVisitor;
 using visitor::SymtabVisitor;
 using visitor::VarUsageVisitor;
 
+using ParamVector = std::vector<std::tuple<std::string, std::string, std::string, std::string>>;
```
Comment (1uc) [inline src/codegen/codegen_coreneuron_cpp_visitor.cpp:42]: Please remove this.

File: src/codegen/codegen_neuron_cpp_visitor.cpp (hunk 2 above)
Comment (JCGoran) [reply src/codegen/codegen_neuron_cpp_visitor.cpp:34]: For some reason, when I remove this, or even just put it in the header file only, the compilation fails (clang 15.0.0 on MacOS), even though it should be visible due to this snippet:
https://github.com/BlueBrain/nmodl/blob/5366532ee2ca2f4961e635672f18043c69e23443/src/codegen/codegen_cpp_visitor.hpp#L266

Anything obvious that I'm missing here?

**Review by alkino — APPROVED** [review]

**Pull request merged**